pkg/natdiscovery/listener.go

```diff
@@ -0,0 +1,73 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package natdiscovery
+
+import (
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	natproto "github.com/submariner-io/submariner/pkg/natdiscovery/proto"
+)
+
+func (nd *natDiscovery) runListener(stopCh <-chan struct{}) error {
+	serverAddress, err := net.ResolveUDPAddr("udp", ":"+strconv.Itoa(natproto.DefaultPort))
+	if err != nil {
+		return errors.Wrap(err, "Error resolving UDP address")
+	}
+
+	serverConnection, err := net.ListenUDP("udp", serverAddress)
```

Comment:
I think we will be listening on both IPv4 and IPv6 addresses. Should we listen just on IPv4?

Reply:
We will need to support ipv6 or mixed environments at some point, so why restrict it or make it specific? :)

Even it if will not help with nat for ipv6, it can help for "reachability" check

Reply:
Currently while creating the localEndpoints, we don't use IPv6 addresses (even if the underlying host has dual-stack configuration). So, listening on IPv6 UDP is not really required since we will not be connecting to the IPv6 addresses in the current implementation.